src/io/stun.rs

```diff
@@ -231,12 +255,10 @@ impl<'a> StunMessage<'a> {
         }
     }
 
+    /// Serialize this message into the provided buffer, returning the final length of the message.
+    ///
+    /// The provided password is used to authenticate the message.
```

Comment:
I love the documentation in this file, thanks!

However I don't want to expose more API than strictly necessary. For your use case, do you need all these functions to be `pub`, or could they be considered an internal detail to str0m's ice module and thus be `pub(crate)`?

Reply:
I actually only need `StunMessage::parse`. I could maybe work around that too if we change the interaction of `accepts_message` and `handle_receive`. I'll propose a patch to get your input.

Reply:
See https://github.com/algesten/str0m/pull/420.

Reply:
Seeing #420, I think I want to backtrack on that idea. `Receive` is an abstraction to help demultiplex STUN, DTLS, RTP, etc by packet inspection. Thinking about the `IceAgent` in isolation, it will never need this demultiplexing – i.e. `Receive` is a "too general" type for `ice`.

I think we should pub expose `StunMessage::parse`, and add the `StunMessage` to the synthetic `ice` module since that it's tightly coupled with the `IceAgent`. The rest of the `StunMessage` fns can be `pub(crate)` though.

Reply:
Okay, so do we want to change `handle_receive` to `StunMessage` too then? We will need an additonal parameter for the address. That one is currently taken out of `net::Receive`.

Reply:
> Okay, so do we want to change `handle_receive` to `StunMessage` too then? We will need an additonal parameter for the address. That one is currently taken out of `net::Receive`.

I am not sure how feasible that is. `stun_server_handle_message` needs everything that is within `net::Receive`:

https://github.com/algesten/str0m/blob/bec55eda9c121b86adc0ee19364f061f533a1294/src/ice/agent.rs#L822

I can make the function not depend on `net::Receive` but then we will just pass `proto`, `source` and `destination` in as separate parameters. That is a fair choice and reduces the dependencies of the the `stun` module. However, the agent already depends on many parts from the `io` module:

https://github.com/algesten/str0m/blob/bec55eda9c121b86adc0ee19364f061f533a1294/src/ice/agent.rs#L7-L10

Thoughts?

Reply:
Ok. I'll look at this tomorrow morning CET when my brain is fresh.

Reply:
> I can make the function not depend on `net::Receive` but then we will just pass `proto`, `source` and `destination` in as separate parameters.

Let me know what you think of https://github.com/algesten/str0m/pull/422.